app/uk/gov/hmrc/customs/declaration/services/StatusResponseFilterService.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.customs.declaration.services
+
+import javax.inject.{Inject, Singleton}
+
+import uk.gov.hmrc.customs.declaration.logging.DeclarationsLogger
+
+import scala.xml._
+
+@Singleton
+class StatusResponseFilterService @Inject() (declarationsLogger: DeclarationsLogger,
+                                             declarationsConfigService: DeclarationsConfigService) {
+
+  val namespacePrefix = "stat"
+  val namespaceBinding = NamespaceBinding(namespacePrefix, "http://gov.uk/customs/declarations/status-request", TopScope)
+  val minimizeEmptyElement = false
+
+  def transform(xml: NodeSeq): NodeSeq = {
+
```

Comment:
I think this could be written as a tail recursive function (for a similar example see https://github.com/hmrc/customs-api-hods-stubs/blob/master/app/uk/gov/hmrc/customs/apihodsstubs/services/statusquery/StatusResponseEditor.scala#L18)
whereby the function take a node, and a List of paths to pull out, and returns the new node. At each stage of the recursion the head is processed and the tail is passed to the next round of recursion.

Reply:
Done